src/Sulu/Bundle/SecurityBundle/EventListener/SuluSecurityListener.php

```diff
@@ -61,7 +71,7 @@ public function onKernelController(FilterControllerEvent $event)
                 $permission = PermissionTypes::VIEW;
                 break;
             case 'POST':
-                if ('postAction' == $controllerDefinition[1]) { // means that the ClassResourceInterface has to be used
+                if (in_array($action, ['postAction', '__invoke'])) { // means that the ClassResourceInterface has to be used
```

Comment:
@nnatter Can be discussed. I personally think if he want to EDIT he should use  `put` or `patch` action instead of a `post`, but at current state I would go with `ADD` as its more common that `post` is a, add and not a edit.  /cc @danrot what do you think

Reply:
I think it is a huge misconception that POST should only be used for create. If you have to add some business logic (as we do in some cases, e.g. when publishing a page) it is totally fine to use POST as well. You can take [Roy Fielding's (the creator of REST) word for it](https://roy.gbiv.com/untangled/2009/it-is-okay-to-use-post).

Regarding `__invoke`: I would say if a controller is defined like that, it gives no clue whatsoever what it is doing. Therefore I would not include any logic about this in here, and let the controller itself define the security if necessary. It's the only place where we really know.

But of course this code should not crash when something unexpected is delivered, so I would **only** fix that.

Reply:
@danrot sure you can use POST for more as create but its more common and I would more connect POST to ADD permisson as to EDIT permission as this is more common.

Reply:
I think I don't really understand the problem here: `postAction` requires an `ADD` permission here, so that should be fine.

I think what @nnatter meant (and I agree with him) is the `__invoke` action. As said, `__invoke` can be anyhting, and the function name does not imply anything about its usage. Therefore I would not do anything in this listener when it gets an action called `__invoke`, except for making sure that it does not break anything.

Reply:
I think even the example code you've posted in the PR description would require only a `VIEW` permission, since it does not add anything, it just returns some values.

Reply:
So your suggestion is to ignore `SecuredControllerInterface` and just do an early return in this case?

Reply:
Hm, either that, or throw a clear error message saying that the `SecuredControllerInterface` only works with classes that also implement the `ClassResourceInterface`. BTW, what is happening currently? :thinking:

Reply:
As I read this dicuseen, I think your ar not talking about the same thing. It's important to consider that's always a combination of HTTP method and method name. A GET and `__invoke` is a `READ`. Only for POST method we need to make assumptions and if somebody implements the `SecuredControllerInterface` it should work.

Reply:
Ah, completely missed the `case 'POST'` part before it :see_no_evil: Now everything makes more sense. I was also mislead by the @nnatter's comment... `__invoke` is not always `POST` with this implementation, right? Then everything is fine :+1:

Reply:
@danrot ok good :)

> BTW, what is happening currently?

Currently callable classes didn't work and did on `$controller[0]` throw a PHP Error trying to access object as array.